src/playlist-maker.js

```diff
@@ -78,6 +78,46 @@ var playlistMaker = function(player, plist) {
     return currentIndex;
   };
 
+  // item can be either
+  //  * a string
+  //  * an array of sources, which are either strings or {src, type} objects
+  //  * a playlist item
+  playlist.contains = function contains(item) {
+    return player.playlist.indexOf(item) !== -1;
+  };
+
+  playlist.indexOf = function indexOf(item) {
+    var ret = -1;
+    var sources;
+    var source;
+    var i;
+
+    if (typeof item === 'string') {
+      ret = indexInSources(list, item);
+    } else {
```

Comment:
I do too but decided not to do so in this function so we only have a single return point. Actually had a bug where I didn't return from all cases and tests failed.